### PR TITLE
@orta => Artsy signup

### LINKED ISSUE #3: Support Artsy sign up
Currently I think it only handles logging in

### PR DESCRIPTION
**Pull request by ashfurrow**
Took a look at the discussion in #3. Not quite sure what you mean – user creation only requires name, email, and password. May later we can have a user update method that accepts dictionaries? 

Fixes #3.

### CODE REVIEW

File: Pod/Classes/ArtsyAuthentication.m [hunk 1]
```diff
@@ -63,18 +62,16 @@ - (void)getUserApplicationXAccessTokenWithEmail:(NSString *)email password:(NSSt
         NSDate *date = [[[ISO8601DateFormatter alloc] init] dateFromString:JSON[@"expires_in"]];
         ArtsyToken *token = [[ArtsyToken alloc] initWithToken:JSON[ArtsyOAuthTokenKey] expirationDate:date];
 
-        strongSelf.router.authToken = token;
```
Comment (ashfurrow) [inline Pod/Classes/ArtsyAuthentication.m:66]: Took this out since it'll be in #6.

**Comment by ashfurrow** [comment]
@orta :green_apple:

**Comment by orta** [comment]
sorry, was meant to get to this yesterday, but this eigen thing took all my attention

File: Pod/Classes/ArtsyAuthentication.h [hunk 2]
```diff
@@ -20,10 +20,10 @@ typedef void (^ArtsyAuthenticationCallback)(ArtsyToken *token, NSError *error);
 - (void)getWeekLongXAppTrialToken:(void (^)(ArtsyToken *token, NSError *error))completion;
 
 /// Gets a 35 year-long authenticated token, nil if network errors
-- (void)getUserApplicationXAccessTokenWithEmail:(NSString *)email password:(NSString *)password :(ArtsyAuthenticationCallback)completion;
+- (void)logInWithEmail:(NSString *)email password:(NSString *)password completion:(ArtsyAuthenticationCallback)completion;
 
 /// Creates a new user, or fails
-- (void)createUserWithUserDictionary:(NSDictionary *)dictionary :(void (^)(NSDictionary *newUserDictionary, NSError *error))completion;
+- (void)createUserWithEmail:(NSString *)email name:(NSString *)name password:(NSString *)password completion:(void (^)(NSDictionary *newUserDictionary, NSError *error))completion;
```
Comment (orta) [inline Pod/Classes/ArtsyAuthentication.h:26]: might want to do both maybe, unless you expect there to be an `updateUserWithDict`? kinda feels like the jurisdiction of this lib-isn, ( well the user manager )

**Comment by orta** [comment]
Looks great.

File: Pod/Classes/ArtsyAuthentication.h (hunk 2 above)
Comment (ashfurrow) [inline Pod/Classes/ArtsyAuthentication.h:26]: Yes exactly – separate out the two and abstract it in the User Manager.

**Pull request merged**